src/components/types/Post.ts

```diff
@@ -0,0 +1,11 @@
+import { User } from './User';
+import { Comment } from './Comment';
+
+export interface Post {
+  userId: number,
+  id: number,
+  title: string,
+  body: string,
+  user: User | null;
+  comments?: Comment[] | null;
```

Comment:
```suggestion
  comments: Comment[];
```
that's the reason you didn't get TS error due to mistake in `App.tsx`

Reply:
Not resolved. You do not need null here